bento/plotting/_lp.py

```diff
@@ -122,20 +121,20 @@ def lp_genes(
     **kwargs
         Options to pass to matplotlib plotting method.
     """
-    lp_stats(data, groupby)
+    lp_stats(sdata, groupby)
 
     palette = dict(zip(PATTERN_NAMES, PATTERN_COLORS))
 
-    n_cells = data.n_obs
-    gene_frac = data.uns["lp_stats"][PATTERN_NAMES] / n_cells
-
-    gene_logcount = data.X.mean(axis=0, where=data.X > 0)
+    n_cells = sdata.table.n_obs
+    gene_frac = sdata.table.uns["lp_stats"][PATTERN_NAMES] / n_cells
+    genes = gene_frac.index
+    gene_logcount = sdata.table[:,genes].X.toarray().mean(axis=0, where=sdata.table[:,genes].X.toarray() > 0)
```

Comment:
pull out gene expression array into variable so it is not queried twice

Reply:
variable = sdata.table[:,genes].X.toarray()